i2c/bmp280_i2c/bmp280_i2c.c

```diff
@@ -0,0 +1,260 @@
+                              struct bmp280_calib_param *params) {
+  float var1, var2;
+  float converted = 0.0;
+
+  var1 = ((float)params->t_fine / 2.0) - 64000.0;
+  var2 = var1 * var1 * ((float)params->dig_p6) / 32768.0;
+  var2 = var2 + var1 * ((float)params->dig_p5) * 2.0;
+  var2 = (var2 / 4.0) + (((float)params->dig_p4) * 65536.0);
+  var1 = (((float)params->dig_p3) * var1 * var1 / 524288.0 +
+          ((float)params->dig_p2) * var1) /
+         524288.0;
+  var1 = (1.0 + var1 / 32768.0) * ((float)params->dig_p1);
+
+  // avoid division by 0
+  if (var1 < 0 || var1 > 0) {
+    converted = 1048576.0 - (float)*pressure;
+    converted = (converted - (var2 / 4096.0)) * 6250.0 / var1;
+    var1 = ((float)params->dig_p9) * converted * converted / 2147483648.0;
+    var2 = converted * ((float)params->dig_p8) / 32768.0;
+
+    converted += (var1 + var2 + ((float)params->dig_p7)) / 16.0;
+    // results in kPa
+    return converted / 1000.0;
+  } else {
+    return 0.0;
+  }
+}
+
+void bmp280_get_calib_params(struct bmp280_calib_param *params) {
+  // raw temp and pressure values need to be calibrated according to
+  // parameters generated during the manufacturing of the sensor
+  // there are 3 temperature params, and 9 pressure params, each with a LSB
+  // and MSB register, so we read from 24 registers
+
+  uint8_t buf[NUM_CALIB_PARAMS] = {0};
+  i2c_write_blocking(i2c_default, (addr & BMP280_WRITE_MODE), &REG_DIG_T1_LSB,
+                     1, true);  // true to keep master control of bus
+  // read in one go as register addresses auto-increment
+  i2c_read_blocking(i2c_default, (addr & BMP280_READ_MODE), buf,
+                    NUM_CALIB_PARAMS, false);  // false, we're done reading
+
+  // store these in a struct for later use
+  params->dig_t1 = (uint16_t)(buf[1] << 8) | buf[0];
+  params->dig_t2 = (int16_t)(buf[3] << 8) | buf[2];
+  params->dig_t3 = (int16_t)(buf[5] << 8) | buf[4];
+
+  params->dig_p1 = (uint16_t)(buf[7] << 8) | buf[6];
+  params->dig_p2 = (int16_t)(buf[9] << 8) | buf[8];
+  params->dig_p3 = (int16_t)(buf[11] << 8) | buf[10];
+  params->dig_p4 = (int16_t)(buf[13] << 8) | buf[12];
+  params->dig_p5 = (int16_t)(buf[15] << 8) | buf[14];
+  params->dig_p6 = (int16_t)(buf[17] << 8) | buf[16];
+  params->dig_p7 = (int16_t)(buf[19] << 8) | buf[18];
+  params->dig_p8 = (int16_t)(buf[21] << 8) | buf[20];
+  params->dig_p9 = (int16_t)(buf[23] << 8) | buf[22];
+}
+
+#endif
+
+int main() {
+  stdio_init_all();
+
+  // useful information for picotool
+  bi_decl(bi_2pins_with_func(PICO_DEFAULT_I2C_SDA_PIN, PICO_DEFAULT_I2C_SCL_PIN,
+                             GPIO_FUNC_I2C));
+  bi_decl(
+      bi_program_description("BMP280 I2C example for the Raspberry Pi Pico"));
+
+#if !defined(i2c_default) || !defined(PICO_DEFAULT_I2C_SDA_PIN) || \
+    !defined(PICO_DEFAULT_I2C_SCL_PIN)
+#warning i2c/bmp280_i2c example requires a board with I2C pins
+  puts("Default I2C pins were not defined");
+#else
+  printf(
```

Comment:
not a fan of splitting a single parameter to a newline

Reply:
I hadn't set up formatting then, I've disabled wrapping now so hopefully this shouldn't happen again